index.js

```diff
@@ -6,20 +6,43 @@
 // https://github.com/ethereum/web3.js
 /////////////////////////////////////////////////////
 
-import AWS from 'aws-sdk';
-import WebsocketProvider  from 'web3-providers-ws';
-import pkg from 'websocket';
-const { w3cwebsocket } = pkg;
+const AWS = require('aws-sdk');
+const WebsocketProvider = require('web3-providers-ws');
+const { w3cwebsocket } = require('websocket');
+
 const Ws = w3cwebsocket;
 
-export default class AWSWebsocketProvider extends WebsocketProvider {
-  constructor(endpoint) {
-    super(endpoint);
-  }
+const validateCredentials = (credentials) => {
+  let valid = true;
+
+  //ensure the provided object has property accessKeyId
+  if (!credentials.hasOwnProperty('accessKeyId')) valid = false
+  //ensure the provided object has property secretAccessKey
+  if (!credentials.hasOwnProperty('secretAccessKey')) valid = false
+  //ensure accessKeyId is not undefined or empty
+  if (credentials.accessKeyId == undefined || credentials.accessKeyId === "") valid = false
+  //ensure secretAccessKey is not undefined or empty
+  if (credentials.secretAccessKey == undefined || credentials.secretAccessKey === "") valid = false
+  
+
+  return valid;
+}
+
+module.exports = class AWSWebsocketProvider extends WebsocketProvider {
+  
+  constructor(url, options) {
+    super(url, options)
+  } 
 
   connect() {
     const region = process.env.AWS_DEFAULT_REGION || 'us-east-1';
-    const credentials = new AWS.EnvironmentCredentials('AWS');
+    let credentials
+    if (this.clientConfig && this.clientConfig.credentials) {
+      if (!validateCredentials(this.clientConfig.credentials)) throw 'Invalid Credentials: Check that your AWS credentials match the standard format';
+      credentials = new AWS.Credentials(this.clientConfig.credentials);
+    } else {
+      credentials = new AWS.EnvironmentCredentials('AWS');
+    }
```

Comment:
Would recommend instead:

```
const creds =
  'clientConfig' in this &&
  this.clientConfig &&
  'credentials' in this.clientConfig &&
  this.clientConfig.credentials;
if (creds && 'accessKeyId' in creds && 'secretAccessKey' in creds) {
  credentials = new AWS.Credentials(this.clientConfig.credentials);
} else {
  credentials = new AWS.EnvironmentCredentials('AWS');
}
```

This also allows you to remove `validateCredentials()`.

Reply:
resolved

Reply:
Resolved - needed to add a check that `clientConfig` is not undefined, as when a user does not provide credentials in the `options.clientConfig` object, validation was failing and the `new AWS.EnvironmentCredentials('AWS');` default was not reached due to an error checking for `credentials in undefined`.